temporal-sdk/src/main/java/io/temporal/internal/client/RootWorkflowClientInvoker.java

```diff
@@ -27,28 +27,52 @@
 import io.temporal.client.WorkflowClientOptions;
 import io.temporal.common.interceptors.WorkflowClientCallsInterceptor;
 import io.temporal.internal.client.external.GenericWorkflowClient;
+import io.temporal.worker.WorkflowTaskDispatchHandle;
 import java.lang.reflect.Type;
 import java.util.*;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
 
 public class RootWorkflowClientInvoker implements WorkflowClientCallsInterceptor {
   private final GenericWorkflowClient genericClient;
   private final WorkflowClientOptions clientOptions;
+  private final EagerWorkflowTaskDispatcher eagerWorkflowTaskDispatcher;
   private final WorkflowClientRequestFactory requestsHelper;
 
   public RootWorkflowClientInvoker(
-      GenericWorkflowClient genericClient, WorkflowClientOptions clientOptions) {
+      GenericWorkflowClient genericClient,
+      WorkflowClientOptions clientOptions,
+      WorkerFactoryRegistry workerFactoryRegistry) {
     this.genericClient = genericClient;
     this.clientOptions = clientOptions;
+    this.eagerWorkflowTaskDispatcher = new EagerWorkflowTaskDispatcher(workerFactoryRegistry);
     this.requestsHelper = new WorkflowClientRequestFactory(clientOptions);
   }
 
   @Override
   public WorkflowStartOutput start(WorkflowStartInput input) {
-    StartWorkflowExecutionRequest request =
-        requestsHelper.newStartWorkflowExecutionRequest(input).build();
-    return new WorkflowStartOutput(genericClient.start(request));
+    try (WorkflowTaskDispatchHandle eagerDispatchHandle = obtainDispatchHandle(input)) {
+      StartWorkflowExecutionRequest.Builder request =
+          requestsHelper.newStartWorkflowExecutionRequest(input);
+      boolean requestEagerExecution = eagerDispatchHandle != null;
+      request.setRequestEagerExecution(requestEagerExecution);
+      StartWorkflowExecutionResponse response = genericClient.start(request.build());
+      WorkflowExecution execution =
+          WorkflowExecution.newBuilder()
+              .setRunId(response.getRunId())
+              .setWorkflowId(request.getWorkflowId())
+              .build();
+      @Nullable
+      PollWorkflowTaskQueueResponse eagerWorkflowTask =
+          requestEagerExecution && response.hasEagerWorkflowTask()
+              ? response.getEagerWorkflowTask()
+              : null;
+      if (eagerWorkflowTask != null) {
+        eagerDispatchHandle.dispatch(eagerWorkflowTask);
```

Comment:
Is there a reason to notify the user if this returns false?

Reply:
No. I made the flag mostly as a "placeholder" for future LocalCompletionHandle / LocalCompletionFuture